Win32-network/src/System/Win32/NamedPipes.hsc

```diff
@@ -147,3 +157,37 @@ foreign import ccall unsafe "windows.h CreateNamedPipeW"
                     -> DWORD
                     -> LPSECURITY_ATTRIBUTES
                     -> IO HANDLE
+
+
+-- | Timeout in milliseconds.
+--
+-- * 'nMPWAIT_USE_DEFAULT_WAIT' indicates that the timeout value passed to
+--   'createNamedPipe' should be used.
+-- * 'nMPWAIT_WAIT_FOREVER' - 'waitNamedPipe' will block forever, until a named
+--   pipe instance is available.
+--
+type TimeOut = DWORD
+#{enum TimeOut,
+ , nMPWAIT_USE_DEFAULT_WAIT = NMPWAIT_USE_DEFAULT_WAIT
+ , nMPWAIT_WAIT_FOREVER     = NMPWAIT_WAIT_FOREVER
+ }
+
+
+-- | Wait until a named pipe instance is available.  If there is no instance at
+-- hand before the timeout, it will error with 'ERROR_SEM_TIMEOUT', i.e.
+-- @invalid argument (The semaphore timeout period has expired)@
+--
+waitNamedPipe :: String  -- ^ pipe name
+              -> TimeOut -- ^ nTimeOut
+              -> IO ()
+waitNamedPipe name timeout =
+    withCString name $ \ c_name ->
+      failIfFalse_  "WaitNamedPipe" $
+        c_WaitNamedPipe c_name timeout
+
+
+-- 'c_WaitNamedPipe' is a blocking call, hence the _safe_ import.
+foreign import ccall safe "windows.h WaitNamedPipeA"
+  c_WaitNamedPipe :: LPCSTR -- lpNamedPipeName
+                  -> DWORD  -- nTimeOut
+                  -> IO Bool
```

Comment:
Surely `IO Bool` isn't right as the result? `Bool` isn't an FFI type.

I think it should be `IO CInt`. 
```
 typedef int BOOL;
```
https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types

Reply:
Yes, it should be `BOOL`.